Add RST documentation

msricher opened the pull request.

msricher committed:
add contact & license doc

msricher committed:
add contact & license doc

FanwangM committed:
Add Jacobian in docs

1. Add Jacobian definition
2. Add example for Jacobian

FanwangM committed:
Add Jacobian in docs

1. Add Jacobian definition
2. Add example for Jacobian

FanwangM committed:
Add header in Jacobian docs

1. Add header
2. Fix typos

FanwangM committed:
Add header in Jacobian docs

1. Add header
2. Fix typos

FanwangM committed:
Add Newton method docs including header

FanwangM committed:
Add Newton method docs including header

FanwangM committed:
Add Gauss–Newton algorithm docs including header

FanwangM committed:
Add Gauss–Newton algorithm docs including header

msricher committed:
fixup docs and add template Makefile

msricher committed:
fixup docs and add template Makefile

matt-chan commented on doc/gauss_newton.rst:
```diff
@@ -0,0 +1,45 @@
+Gauss–Newton Algorithm
+======================
+
+The advantage of Gauss-Newton Method is that the second derivatives are not necessarily needed to be
```
do not necessarily need

matt-chan commented on doc/gauss_newton.rst:
```diff
@@ -0,0 +1,45 @@
+======================
+
+The advantage of Gauss-Newton Method is that the second derivatives are not necessarily needed to be
+computed.The limitation is also obvious since it can only handle with the minimization the sum of
```
- space before The. 
- "since it can only minimize the sum of squared function values".

fwmeng88 replied:
1. Why is space needed? When I compile the website, it looks like something unusual.
2. Fixed.

matt-chan commented on doc/gauss_newton.rst:
```diff
@@ -0,0 +1,45 @@
+
+The advantage of Gauss-Newton Method is that the second derivatives are not necessarily needed to be
+computed.The limitation is also obvious since it can only handle with the minimization the sum of
+squared function values. The Gaussian-Newton is mainly used to resolve the non-linear least squares
```
Gauss-Newton method

fwmeng88 replied:
Fixed.

matt-chan commented on doc/gauss_newton.rst:
```diff
@@ -0,0 +1,45 @@
+The advantage of Gauss-Newton Method is that the second derivatives are not necessarily needed to be
+computed.The limitation is also obvious since it can only handle with the minimization the sum of
+squared function values. The Gaussian-Newton is mainly used to resolve the non-linear least squares
+problems, named after two famous mathematicians Carl Friedrich Gauss and Isaac Newton.
```
problems. It is named after the two

fwmeng88 replied:
Fixed.

matt-chan commented on doc/gauss_newton.rst:
```diff
@@ -0,0 +1,45 @@
+.. math::
+    x_{s+1} = x_s - {(J^{\top} J)}^{-1} {J^{\top}} f(x_s)
+
+where an initial guess :math:`x_0`. When the matrix :math:`f` is a squared one, the simplified
```
- where :math:`x_0` is the initial guess.
- is squared

fwmeng88 replied:
Fixed.

matt-chan commented on doc/jacobian.rst:
```diff
@@ -0,0 +1,100 @@
+Definition
+----------
+
+Let's suppose we have a vector valued function of a vector
```
- Delete Let's
- Delete of a vector

fwmeng88 replied:
Done.

matt-chan commented on doc/jacobian.rst:
```diff
@@ -0,0 +1,100 @@
+
+Let's suppose we have a vector valued function of a vector
+:math:`f: \mathbb{R}^n \rightarrow \mathbb{R}^m`. This function will take the input vector
+:math:`x \in \mathbb{R}^n` and maps it to
```
map

fwmeng88 replied:
Done.

matt-chan commented on doc/jacobian.rst:
```diff
@@ -0,0 +1,100 @@
+Let's suppose we have a vector valued function of a vector
+:math:`f: \mathbb{R}^n \rightarrow \mathbb{R}^m`. This function will take the input vector
+:math:`x \in \mathbb{R}^n` and maps it to
+the result vector :math:`f(x) \in \mathbb{R}^m`. The Jacobian is can be also termed as
```
The Jacobian is also called

fwmeng88 replied:
Done

matt-chan commented on doc/jacobian.rst:
```diff
@@ -0,0 +1,100 @@
+    \end{bmatrix}
+    }
+
+Or put it in a element-wise way,
```
Or, element-wise,

fwmeng88 replied:
Done

matt-chan commented on doc/jacobian.rst:
```diff
@@ -0,0 +1,100 @@
+------
+
+In our package, we have two different ways to approximate Jacobian, namely the forward difference
+method and the central difference one. All these two methods are based on the principle that the
```
- s/one/method/
- s/All/both/
- s/the//

matt-chan commented on doc/newton.rst:
```diff
@@ -0,0 +1,46 @@
+Newton Algorithm
+================
+
+Newton method (also knows as the Newton-Raphson method, Newton's iteration) is a algorithm that utilizes the first terms of Taylor Polynomials of function :math:`f(x)` to find the roots of
```
- Newton's method
- s/a/an/
- s/of Taylor Polynomials/the taylor series/

fwmeng88 replied:
Done

matt-chan commented on doc/newton.rst:
```diff
@@ -0,0 +1,46 @@
+================
+
+Newton method (also knows as the Newton-Raphson method, Newton's iteration) is a algorithm that utilizes the first terms of Taylor Polynomials of function :math:`f(x)` to find the roots of
+:math:`f(x)=0`. For the Taylor series of :math:`f(x)` about the pint
```
point

fwmeng88 replied:
Done.

matt-chan commented on doc/newton.rst:
```diff
@@ -0,0 +1,46 @@
+
+Newton method (also knows as the Newton-Raphson method, Newton's iteration) is a algorithm that utilizes the first terms of Taylor Polynomials of function :math:`f(x)` to find the roots of
+:math:`f(x)=0`. For the Taylor series of :math:`f(x)` about the pint
+:math:`x_{s+1} = x_s + \epsilon` where the :math:`s` and :math:`s+1` are indexes,
```
s/the//

fwmeng88 replied:
done

matt-chan commented on doc/newton.rst:
```diff
@@ -0,0 +1,46 @@
+.. math::
+    f(x_{s+1}) \approx f(x_s) + f^{\prime}({x_s}) \epsilon
+
+Because we are solving :math:`f(x)=0`, then for Newton method, the problem can be formulated as
```
Newton's method

fwmeng88 replied:
Done

matt-chan reviewed (changes requested):
Grammar fixes needed. I didn't check the math because github doesn't render it for some reason.

Commit:
Add Jacobian class

Commit:
Add Jacobian for inheritance

gabrielasd committed:
Add Broyden methods

gabrielasd committed:
Work on Broyden functions

gabrielasd committed:
Make division by vector norm before generating matrix in Broythen methods

gabrielasd committed:
Add Broyden methods tests

xiaominhc committed:
Added step and update files

gabrielasd committed:
Work on docstrings of Broyden update methods

Commit:
Add call function to Jacobian class

Commit:
Ported args from FiniteDIffJacobian to super class

Commit:
Add docstrings to Jacobian class

Commit:
Merge broyden methods with jacobian_class

Commit:
Add docstring for Jacobian class

xiaominhc committed:
added option method (works)

gabrielasd committed:
Replace f_k with delta_f in Broyden update functions

gabrielasd committed:
Fix tests Broyden methods

Commit:
Move arguments back to original class

Commit:
Merge branch 'broyden_methods' of https://github.com/msricher/optimizer into jacobian_class

Commit:
Add input tests for Jacobian

xiaominhc committed:
Generalized nonlinear newton iteration

xiaominhc committed:
Merge branch 'jacobian_class' of github.com:msricher/optimizer into addoptions

xiaominhc committed:
Modified update_analytic

xiaominhc committed:
Cleaned step.py, it is now a function, not a class

gabrielasd committed:
Generalize input arguments of Broyden methods

gabrielasd committed:
Fix Broyden updates tests

FanwangM committed:
Fix grammar mistakes in gauss_newton.rst

FanwangM committed:
Fix grammar mistakes in gauss_newton.rst

FanwangM committed:
Fix grammar erros in jacobian docs

FanwangM committed:
Fix grammar erros in jacobian docs

FanwangM committed:
Fix grammar errors in newton doc

FanwangM committed:
Fix grammar errors in newton doc

FanwangM committed:
Fix assert_raises to be ValueError

Fix some of the TypeError to be ValueError. For example, "assert_raises(ValueError, newton_solve, f1, x_0, J=j1, maxiter=-1)" should be a value error, beacuse the maxiter=-1 is the right data type, but not right value.

xiaominhc committed:
Remove input checks in step

Commit:
Change jacobian constructor to being fixed

Commit:
Change jacobian constructor to being fixed

xiaominhc committed:
Added test for broyden methods, but they dont work

xiaominhc committed:
Added test for broyden methods, but they dont work

FanwangM committed:
Get back the working version in newton test

Commit:
Fix pydocstyle issues

Commit:
Fix merge conflict

Commit:
Fix blank line at end of file

xiaominhc committed:
Tests work now, passes pydoc and pycodestyle

xiaominhc committed:
Tests work now, passes pydoc and pycodestyle

Commit:
Fix variable initialization in newton

Commit:
Fix variable initialization in newton

Commit:
wow

Commit:
wow

Commit:
Fix Bad Broyden

Commit:
Fix Bad Broyden

Commit:
wow again

Commit:
wow again

Commit:
Add changes to newton file to match new notation

Commit:
Add changes to newton file to match new notation

Commit:
Fix good broyden

Commit:
Fix good broyden

Commit:
Fix new notation to match iteration=1 case

Commit:
Fix new notation to match iteration=1 case

Commit:
Fix tests for good broyden

Commit:
Fix tests for good broyden

Commit:
Add input tests to gauss newton

Commit:
Add input tests to gauss newton

Commit:
Add __all__ to jacobian test

Commit:
Add __all__ to jacobian test

Commit:
Fix merge conflicts

Commit:
Fix test doc string

Commit:
Fix doc string error in test_jacobian

Commit:
Fix good and bad broyden

Commit:
Fix broyde methods

Commit:
Add test to gauss newton for using approximation jacobian

Commit:
Add doc string to the test

Commit:
Add test for broyden method in jacobian

Commit:
Fix doc string error

Commit:
Fix pylint errors

Commit:
Merge pull request #15 from msricher/changetests

Changetests

xiaominhc committed:
Added SR1 update to J and Jinv (untested)

FanwangM committed:
Add BFGS jacobian update

Add both BFGS jacobian update and its test file

gabrielasd committed:
Add BFGS and SR1 options to newton

gabrielasd committed:
Add BFGS and SR1 options to newton

xiaominhc committed:
Correct and add tests for sr1 and sr1_inv

Commit:
Add dfp and inverse method with tests

Commit:
Remove DFP inverse

gabrielasd committed:
Add DFP options to newton

gabrielasd committed:
Fix merge conflicts

Commit:
Fix merge conflicts

Commit:
Fix test

Commit:
Merge BFGS

Commit:
Add DFP tests

Commit:
Fix merge

gabrielasd committed:
Add quasi-Newton docs

FanwangM committed:
Add mathematical equations in method doctrines

1. Add equations in each Jacobian update method
2. Add full name for some abbreviations in docstrings

FanwangM committed:
Merge branch 'docs' of https://github.com/msricher/optimizer into docs

FanwangM committed:
Add spaces for parameter descriptions

gabrielasd committed:
Work on quasi-Newton docs

gabrielasd committed:
Merge branch 'quasinewton_doc' into docs

msricher commented:
Closed; branch was messed up
